### PR TITLE
Add findAll to Schemas

### PR DESCRIPTION
**Pull request by jensdev**

### CODE REVIEW

File: src/services/data/schemas.ts [hunk 1]
```diff
@@ -53,6 +53,38 @@ export default (client, httpAuth: HttpInstance): DataSchemasService => ({
     };
   },
 
+  /**
+   * Request a list of all schemas
+   * Do not pass in an rql with limit operator!
+   * Permission | Scope | Effect
+   * - | - | -
```
Comment (ThomasReyskens) [inline src/services/data/schemas.ts:60]: add newlines to improve jsdocs

Comment (galmail) [reply src/services/data/schemas.ts]: done!

File: src/rql.ts [hunk 2]
```diff
@@ -114,7 +114,10 @@ export function rqlBuilder(rql?: RQLString): RQLBuilder {
         typeof value === 'string' ? value : value.join(',')
       );
     },
-    limit(limit, offset?) {
+    limit(limit, offset) {
```
Comment (ThomasReyskens) [inline src/rql.ts:117]: why not optional?

File: src/services/data/schemas.ts [hunk 3]
```diff
@@ -36,6 +36,28 @@ export default (client, httpAuth: HttpInstance): DataSchemasService => ({
...
   },
 
+  async findAll(options?: { rql?: RQLString }): Promise<Schema[]> {
+    // Extra check is needed because this function is call recursively with updated RQL
+    // But on the first run, we need to set the limit to the max to optimize
+    const result: PagedResult<Schema> = await this.find({
+      rql:
+        options?.rql && options.rql.includes('limit(')
+          ? options.rql
+          : rqlBuilder(options?.rql).limit(50).build(),
```
Comment (ThomasReyskens) [inline src/services/data/schemas.ts:46]: const DEFAULT_LIMIT = 50

Comment (ThomasReyskens) [reply src/services/data/schemas.ts:46]: add console.log('WARNING: total amount is > 2000, be aware that this function can hog up resources')

**Review by ThomasReyskens — APPROVED** [review]

**Pull request merged**